src/authentication/authentication-hapi-plugin.ts

```diff
@@ -106,9 +94,37 @@ class AuthenticationHapiPlugin extends HapiPlugin {
 
   public async getTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
     try {
-      const teamId = parseInt(request.paramsArray[0], 10);
-      const teamToken: TeamToken[] = await teamTokenQuery(this.db, undefined, teamId);
-      if (!teamToken.length) {
+      const credentials = request.auth.credentials as AccessToken;
+      const userName = sanitizeUsername(credentials.sub);
+      const isAdmin = await this.isAdmin(userName);
+      const userTeams = await this.gitlab.getUserGroups(userName);
+      const parameterKey = 'teamIdOrName';
+      const teamIdOrName = request.params[parameterKey];
+      let ownTeam: Group | undefined;
+      if (userTeams && teamIdOrName) {
+        ownTeam = userTeams.find(assertTeam.bind(undefined, teamIdOrName));
```

Comment:
Since assertTeam seems to be used mainly in this fashion (i.e. return a function that takes in a team and tells whether it's what we're looking for or not), having the definition be the following is, to me, clearer about the intended use:

```javascript
export function assertTeam(teamNameOrId: string |  number) {
  return (team: Group) => team.name.toLowerCase() === teamNameOrId.toString()
    || team.path === teamNameOrId.toString()
    || team.id === parseInt(teamNameOrId.toString(), 10);
}
```

Additionally, `assertTeam(teamIdOrName)` is somewhat simpler to understand than `assertTeam.bind(undefined, teamIdOrName)`.